internal/controller/pause.go

```diff
@@ -115,12 +114,27 @@ func (pm *PauseModule) updatePipelineLifecycle(ctx context.Context, restConfig *
 		return err
 	}
 
+	// TODO: I noticed if any of these fields are nil, this function errors out - but can't remember why they'd be nil
 	err = unstructured.SetNestedField(unstruc.Object, status, "spec", "lifecycle", "desiredPhase")
 	if err != nil {
 		return err
 	}
 
-	return kubernetes.UpdateUnstructuredCR(ctx, restConfig, unstruc, common.PipelineGVR, pipeline.Namespace, pipeline.Name)
+	resultObj, err := kubernetes.UnstructuredToObject(unstruc)
+	if err != nil {
+		return err
+	}
+	if resultObj == nil {
+		return fmt.Errorf("error converting unstructured %+v to object, result is nil?", unstruc.Object)
+	}
+
+	err = kubernetes.UpdateCR(ctx, restConfig, resultObj, "pipelines")
+	if err != nil {
+		return err
+	}
+	*pipeline = *resultObj
+	return nil
```

Comment:
By calling `UpdateCR()`, it will update `resultObj` to the returned value from K8S and we can then set `pipeline` to it, meaning any subsequent update call that might get made will get the latest value and won't incur a resource version conflict.